File: src/main/perl/consumer.pl
  use Net::Stomp;
  use XML::Simple;



  my $stomp = Net::Stomp->new( { hostname => 'localhost', port => '61613' } );
  $stomp->connect( { login => 'system', passcode => 'manager' } );

  $stomp->subscribe(
      {   destination             => '/topic/STOCKS.JAVA',
          'ack'                   => 'client',
          'activemq.prefetchSize' => 1
      }
  );
  $stomp->subscribe(
      {   destination             => '/topic/STOCKS.IONA',
          'ack'                   => 'client',
          'activemq.prefetchSize' => 1
      }
  );


  my $count = 0;

  while ($count++ < 100) {
    my $frame = $stomp->receive_frame;
    my $xml = XMLin($frame->body);
    print $xml->{name} . "\t" . sprintf("%.2f", $xml->{price}) . "\t";
    print sprintf("%.2f", $xml->{offer}) . "\t" . ($xml->{up} eq 'true' ? 'up' : 'down') . "\n";

    $stomp->ack( { frame => $frame } );
  }

  $stomp->disconnect;

File: src/main/php/consumer.php
<?

require_once('Stomp.php');

$stomp = new Stomp("tcp://localhost:61613");

$stomp->connect('system', 'manager');

$stomp->subscribe("/topic/STOCKS.JAVA");
$stomp->subscribe("/topic/STOCKS.IONA");

$i = 0;
while($i++ < 100) {

    $frame = $stomp->readFrame();
    $xml = new SimpleXMLElement($frame->body);
    echo $xml->attributes()->name . "\t" . number_format($xml->price,2)
       . "\t" . number_format($xml->offer,2)
       . "\t" . ($xml->up == "true"?"up":"down") . "\n";
    $stomp->ack($frame);

}

$stomp->disconnect();


?>
